src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -504,6 +504,26 @@ bool generate_ansi_c_start_function(
 
   record_function_outputs(symbol, init_code, symbol_table);
 
+  // now call destructor functions (a GCC extension)
+
+  for(const auto &symbol_table_entry : symbol_table.symbols)
```

Comment:
Does this need to be guarded so that it only happens in gcc-mode ? Or is that decision handled elsewhere, at the point where ID_destructor is applied?

Reply:
And one other sanity check - GCC docs suggest that destructors can be given an (optional) priority number which controls the order in which they are executed. It looks like that is not yet supported?

Reply:
> Does this need to be guarded so that it only happens in gcc-mode ? Or is that decision handled elsewhere, at the point where ID_destructor is applied?

The parser won't accept those attributes when not in GCC mode, so that's where I'd argue this distinction is handled. If people want to forego the front-end and create symbol tables themselves (e.g., as input to symtab2gb) then it should be ok to let them use destructors in this way.